### PR TITLE
feat: add sp metrics for internal/external sp   

### PR DESCRIPTION
**Pull request by randyahx**
### Description

Add sp metrics for internal/external sp to better identify the source of the errors.  

### Rationale

To segregate the issues and create alerts for them.  

### Example

none  

### Changes

Notable changes: 
none

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -301,7 +301,7 @@ func (e *Executor) QueryChallengeSlashCoolingOffPeriod() (uint64, error) {
 	client := e.clients.GetClient()
 	params, err := client.ChallengeParams(context.Background(), &challengetypes.QueryParamsRequest{})
 	if err != nil {
-		logging.Logger.Errorf("query challenge params failed, err=%+v", err.Error())
+		logging.Logger.Errorf("query slash cooling off period failed, err=%+v", err.Error())
```
Comment (forcodedancing) [inline executor/executor.go:304]: Why change this comment?

File: verifier/hash_verifier.go [hunk 2]
```diff
@@ -307,7 +314,29 @@ func (v *Verifier) compareHashAndUpdate(challengeId uint64, chainRootHash []byte
 		return err
 	}
 	// update metrics if no err
+	logging.Logger.Infof("challenge succeeded for challengeId: %d, hash mismatched", challengeId)
 	v.metricService.IncVerifiedChallenges()
 	v.metricService.IncChallengeSuccess()
 	return err
 }
+
+func (v *Verifier) isInternalSP(spEndpoint string) bool {
```
Comment (forcodedancing) [inline verifier/hash_verifier.go:323]: We can use a map to make it more efficient.

File: verifier/hash_verifier.go [hunk 3]
```diff
@@ -307,7 +314,29 @@ func (v *Verifier) compareHashAndUpdate(challengeId uint64, chainRootHash []byte
...
+
+func (v *Verifier) updateDeduplicationIntervalLoop() {
+	ticker := time.NewTicker(UpdateDeduplicationInterval)
+	for range ticker.C {
+		updatedDeduplicationInterval, err := v.executor.QueryChallengeSlashCoolingOffPeriod()
+		if err != nil {
+			logging.Logger.Errorf("error updating deduplication interval, err=%s", err.Error())
+			return
+		}
+		v.deduplicationInterval = updatedDeduplicationInterval
```
Comment (forcodedancing) [inline verifier/hash_verifier.go:340]: We may need a mutex or lock here.

Comment (forcodedancing) [reply verifier/hash_verifier.go:340]: and where `v.deduplicationInterval` been read.

Comment (randyahx) [reply verifier/hash_verifier.go:341]: fixed

File: verifier/hash_verifier.go (hunk 2 above)
Comment (randyahx) [reply verifier/hash_verifier.go:323]: this would require the endpoints to match exactly

**Review by forcodedancing — APPROVED** [review]

**Pull request merged**